File: back-end/src/modules/user/mappers/user.mapper.ts
/*
 * @description: user mapper
 * @author: Nhut Tan
 * @date: 2025-09-08
 * @modified: 2025-09-12
 * @version: 1.0.1
 * */

import { UserEntity } from '../entities/user.entity';
import { UserEntityResponseDto } from '../dtos/user-entity-response.dto';
import { Injectable } from '@nestjs/common';

@Injectable()
export class UserMapper {
  toUserResponseDto(user: UserEntity): UserEntityResponseDto {
    return {
      id: user.id,
      username: user.username,
      email: user.email,
      role: user.role.name,
      status: user.status.toString(),
      createdAt: user.createdAt.toString(),
      updatedAt: user.updatedAt.toString(),
    };
  }
}

File: ECommerce/admin/src/main/java/com/example/admin/mapper/UserMapper.kt
package com.example.admin.mapper

import com.example.admin.dto.UserDTO
import com.example.admin.model.UserModel

object UserMapper {

    fun toUserModel(dto: UserDTO): UserModel {
        return UserModel(
                id = dto.id,
                username = dto.username,
                email = dto.email,
                fullName = dto.fullName,
                role = dto.role?.name ?: "Unknown",
                status = dto.status,
                createdAt = dto.createdAt ?: "",
                roleId = dto.role?.id
        )
    }

    fun toUserList(dtos: List<UserDTO>): List<UserModel> {
        return dtos.map { toUserModel(it) }
    }
}